databuilder/databuilder/extractor/kafka_schema_registry_extractor.py

```diff
@@ -0,0 +1,215 @@
+                    password=self._registry_password
+                )
+            )
+        else:
+            self._client = SchemaRegistryClient(
+                url=self._registry_base_url,
+            )
+
+        self._check_registry_connection()
+
+        self._extract_iter: Union[None, Iterator] = None
+
+    def extract(self) -> Union[TableMetadata, None]:
+        if not self._extract_iter:
+            self._extract_iter = self._get_extract_iter()
+        try:
+            return next(self._extract_iter)
+        except StopIteration:
+            return None
+        except Exception as e:
+            logger.error(f'Failed to generate next table: {e}')
+            return None
+
+    def get_scope(self) -> str:
+        return 'extractor.kafka_schema_registry'
+
+    def _get_extract_iter(self) -> Optional[Iterator[TableMetadata]]:
+        """
+        Return an iterator generating TableMetadata for all of the schemas.
+        """
+        for schema_version in self._get_raw_extract_iter():
+            subject = schema_version.subject
+            schema = schema_version.schema.raw_schema
+            LOGGER.info((f'Subject: {subject}, '
+                         f'Schema: {schema}'))
+
+            try:
+                yield KafkaSchemaRegistryExtractor._create_table(
+                    schema=schema,
+                    subject_name=subject,
+                    cluster_name=schema.get(
+                        'namespace', 'kafka-schema-registry'
+                    ),
+                    schema_name=schema.get('name', ''),
+                    schema_description=schema.get('doc', None),
+                )
+            except Exception as e:
+                logger.warning(f'Failed to generate table for {subject}: {e}')
+                continue
+
+    def _get_raw_extract_iter(self) -> Iterator[SchemaVersion]:
+        """
+        Return iterator of results row from schema registry
+        """
+        subjects = self._get_all_subjects()
+
+        LOGGER.info(f'Number of extracted subjects: {len(subjects)}')
+        LOGGER.info(f'Extracted subjects: {subjects}')
+
+        for subj in subjects:
+            subj_schema = self._get_schema(subj)
+            LOGGER.info(f'Subject <{subj}> max version: {subj_schema.version}')
+
+            yield subj_schema
+
+    def _get_schema(self,
+                    subject: str) -> SchemaVersion:
+        """
+        Return the schema of the given subject
+        """
+        return self._client.get_schema(subject)
+
+    def _get_all_subjects(self) -> List[str]:
+        """
+        Return all subjects from Kafka Schema registry
+        """
+        return self._client.get_subjects()
+
+    def _check_registry_connection(self) -> None:
+        """
+        Check to see if the connection to Schema Registry with provided
+        Authentication is okay or not.
+        """
+        try:
+            self._client.get_subjects()
```

Comment:
this seems like overkill for checking connection (for example we have 500+ schemas in sr and this would pull them just for checking connection). maybe we would be good not having _check_registry_connection at all? assuming that we handle exceptions properly whenever we interact with schema registry it's not really needed

Reply:
You are right. What about just getting one subject? I think it will not be bad to have a _check_registry_connection but please let me know what do you think about it so I can change it like that.